openapi.yaml

```diff
@@ -0,0 +1,730 @@
+          required: false
+          schema:
+            type: number
+      responses:
+        '200':
+          description: Services fetched successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ServicePagination'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error
+      security:
+        - api_key: []
+  /services/{serviceId}:
+    get:
+      tags:
+        - services
+      summary: Retrieve service
+      description: Retrieve a service by `serviceId`
+      operationId: getService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Service fetched successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error
+      security:
+        - api_key: []
+    put:
+      tags:
+        - services
+      summary: Update service
+      description: |
+        Update an existing service by by `serviceId`.
+      operationId: updateService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Updated service payload
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServicePayload'
+        required: true
+      responses:
+        '200':
+          description: Service updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error
+      security:
+        - api_key: []
+    delete:
+      tags:
+        - services
+      summary: Delete service
+      description: |
+        Delete a service by `serviceId`.<br>_Note: This API operation needs the __Manage
+        API key__ owned by the service user, otherwise `403 forbidden` will be returned to the caller._
```

Comment:
@giuseppedipinto I just wanted to avoid referring to parameter names in the operation description. It's just a decoupling from what (description) and how (parameter, request/response body, ecc.) the operation does. To me, description is more a high level concept.
E.g.: we might have to change path param name from  _serviceId_ to _id_ and, with this "coupling", we will have to change also the operation description even if nothing has changed from that "side" (is not a change in behavior). I hope this example clarifies what i mean.

Reply:
@rodoviti you have been very clear and I don't see any problems in changing the form of the description, the substance does not change but as you rightly say, let's avoid direct references to parameters.
Let's say you almost convinced me! :-)